catroid/src/androidTest/java/org.catrobat.catroid/uiespresso/util/matchers/AdapterViewMatchers.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.util.matchers;
+
+import android.view.View;
+import android.widget.Adapter;
+import android.widget.AdapterView;
+
+import org.catrobat.catroid.common.ProjectData;
+import org.catrobat.catroid.ui.adapter.ProjectListAdapter;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+public final class AdapterViewMatchers {
+	private AdapterViewMatchers() {
+		throw new AssertionError();
+	}
+
+	public static Matcher<View> withAdaptedDataAt(final String expectedContent, final int listPosition) {
+		return new TypeSafeMatcher<View>() {
+
+			@Override
+			public void describeTo(Description description) {
+			}
+
+			@Override
+			public boolean matchesSafely(View view) {
+				if (!(view instanceof AdapterView)) {
+					return false;
+				}
+
+				Adapter adapter = ((AdapterView) view).getAdapter();
+				String listItem;
+				if (adapter instanceof ProjectListAdapter) {
+					listItem = ((ProjectData) adapter.getItem(listPosition)).projectName;
+				} else {
+					listItem = (String) adapter.getItem(listPosition);
+				}
+
+				return expectedContent.equals(listItem);
+			}
+		};
+	}
+
+	public static Matcher<View> withIndex(final Matcher<View> matcher, final int index) {
+		return new TypeSafeMatcher<View>() {
+			int currentIndex = 0;
+
+			@Override
+			public void describeTo(Description description) {
+				description.appendText("with index: ");
+				description.appendValue(index);
+				matcher.describeTo(description);
+			}
+
+			@Override
+			public boolean matchesSafely(View view) {
+				return matcher.matches(view) && currentIndex++ == index;
```

Comment:
I'm not quite sure about when exactly matchesSafely is called (you are the Espresso Guru :)), but it's a member variable. So I think it might be to iterate through a list, for example. In the first pass it's 0, then 1 etc. until it matches..
 (I've taken and adapted that matcher from https://stackoverflow.com/questions/29378552/in-espresso-how-to-choose-one-the-view-who-have-same-id-to-avoid-ambiguousviewm)
So, I'm not quite sure, but I'll try without incrementing whether the test passes